GraphAlgorithms/LargestCommonSubgraphNaiveFinder.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GraphAlgorithms
+{
+    public static class LargestCommonSubgraphNaiveFinder
+    {
+        public static LargestCommonSubgraphFinderResult Find(Graph graph1, Graph graph2)
+        {
+            int minSize = Math.Min(graph1.Size, graph2.Size);
+            int maxSize = Math.Max(graph1.Size, graph2.Size);
+            Graph result = new Graph(maxSize);
+            int resultNumOfEdges = 0;
+            if (graph1.Size < graph2.Size)
+                (graph1, graph2) = (graph2, graph1);
+            List<int> listOfIndices = Enumerable.Range(0, graph1.Size).ToList();
+            List<int> rowPermutationUsedInResult = Enumerable.Range(0, graph1.Size).ToList();
+            List<int> columnPermutationUsedInResult = Enumerable.Range(0, graph1.Size).ToList();
+            List<List<int>> indicesPermutations = GeneratePermutations(listOfIndices);
+            foreach (var columnPermutation in indicesPermutations)
+            {
+                foreach (var rowPermutation in indicesPermutations)
+                {
+                    // search for potential result
+                    Graph potentialResult = new Graph(maxSize);
+                    int numOfEdges = 0;
+                    for (int i = 0; i < minSize; i++)
+                    {
+                        for (int j = 0; j < minSize; j++)
+                        {
+                            // remove edeges from a vertex to itself
+                            if (i == j) continue;
+                            var graph1Value = graph1.AdjacencyMatrix[rowPermutation[i], columnPermutation[j]];
+                            var graph2Value = graph2.AdjacencyMatrix[i, j];
+                            if (graph1Value != 0 && graph2Value != 0)
```

Comment:
It looks like there is always at most one edge in the test graphs, but in general we could have multiple, so wouldn't the condition be: `graph1Value != 0 && graph2Value != 0 && graph1Value == graph2Value`?
And then the result could be updated like that `potentialResult.SetNumberOfEdges(i, j, graph1Value);`

Reply:
Also then we would have to modify `GetGraphAfterPermutations` slightly